src/filesystem.rs

```diff
@@ -34,11 +34,12 @@
 
 use crate::{
     conf,
-    vfs::{self, VFS},
+    vfs::{self, OverlayFS, VFS},
     Context, GameError, GameResult,
 };
 use directories::ProjectDirs;
-use std::{env, io, io::SeekFrom, path};
+use parking_lot::Mutex;
```

Comment:
why use `parking_lot::Mutex` instead of a standard mutex?

Reply:
std mutex gained good performance on linux only recently but parking_lot still has better "low contention" performance, I don't expect mutexes to ever block (realized an issue here, mutexes won't work in wasm) so there will always be zero contention. And I always see parking_lot being recommended over std in general but maybe it doesn't make sense to bring in another dependency (and even if we did, it should probably be applied to all mutexes in that case)

Reply:
actually nvm parking_lot mutexes do work in wasm as long as they don't block, which they won't if there are no threads, which there aren't in wasm (says some github comment, didn't verify)

Reply:
I'll switch to std mutex for now though, we can decide on that later

Reply:
> actually nvm parking_lot mutexes do work in wasm as long as they don't block, which they won't if there are no threads, which there aren't in wasm (says some github comment, didn't verify). Do std mutexes work there though?

Looking it up quickly it seems they do. I'd say we stick with `std` until we get a good reason to switch and then switch everywhere at once, yes.